skyportal/utils/offset.py

```diff
@@ -285,15 +286,23 @@ def _calculate_best_position_for_offset_stars(
         df = df[df["offset_arcsec"] < sigma_clip * np.std(df["offset_arcsec"])]
 
     # TODO: add the ability to use only use observations from some filters
-    if how == "snr2":
-        df["snr"] = df["flux"] / df["fluxerr"]
-        diff_ra = np.average(df["ra_offset"], weights=df["snr"] ** 2)
-        diff_dec = np.average(df["dec_offset"], weights=df["snr"] ** 2)
-    elif how == "invvar":
-        diff_ra = np.average(df["ra_offset"], weights=1 / df["ra_unc"] ** 2)
-        diff_dec = np.average(df["dec_offset"], weights=1 / df["dec_unc"] ** 2)
-    else:
-        log(f"Warning: do not recognize {how} as a valid way to weight astrometry")
+    try:
+        if how == "snr2":
+            df["snr"] = df["flux"] / df["fluxerr"]
+            diff_ra = np.average(df["ra_offset"], weights=df["snr"] ** 2)
+            diff_dec = np.average(df["dec_offset"], weights=df["snr"] ** 2)
+        elif how == "invvar":
+            diff_ra = np.average(df["ra_offset"], weights=1 / df["ra_unc"] ** 2)
+            diff_dec = np.average(df["dec_offset"], weights=1 / df["dec_unc"] ** 2)
+        else:
+            log(f"Warning: do not recognize {how} as a valid way to weight astrometry")
+            return (med_ra, med_dec)
+    except ZeroDivisionError as e:
+        log(f"ZeroDivisionError in calculating position with {how}: {e}")
```

Comment:
Do we want to do something special in SourceFinderHandler when these errors occur?  If so, this function may want to raise.

Reply:
No, the median position will be fine enough. It's not an error they needs to be dealt with at the user level. We're already in the "likely doesn't matter" regime with the median vs the weighted average pos.

Reply:
OK, great